libcst/matchers/_matcher_base.py

```diff
@@ -789,26 +869,61 @@ def _attribute_matches(  # noqa: C901
     # types were ignored.
     return _matches(
         cast(Union[MaybeSentinel, RemovalSentinel, libcst.CSTNode], node),
-        cast(Union[BaseMatcherNode, MatchIfTrue], matcher),
+        cast(Union[BaseMatcherNode, MatchIfTrue, MatchMetadata], matcher),
+        metadata_lookup,
     )
 
 
+def _metadata_matches(
+    node: libcst.CSTNode,
+    metadata: Union[
+        MatchMetadata,
+        AllOf[MatchMetadata],
+        OneOf[MatchMetadata],
+        InverseOf[MatchMetadata],
+    ],
+    metadata_lookup: Callable[[meta.ProviderT, libcst.CSTNode], object],
+) -> bool:
+    if isinstance(metadata, OneOf):
+        return any(
+            _metadata_matches(node, metadata, metadata_lookup)
+            for metadata in metadata.options
+        )
+    elif isinstance(metadata, AllOf):
+        return all(
+            _metadata_matches(node, metadata, metadata_lookup)
+            for metadata in metadata.options
+        )
+    elif isinstance(metadata, InverseOf):
+        return not _metadata_matches(node, metadata.matcher, metadata_lookup)
```

Comment:
According to Codecov, we don't have test coverage of any of these clauses, only the last one? If that's true, it seems worth fixing; this is pretty core to the functionality.

Reply:
I'll check it out, but I thought I had tests for this.